platform/pulpcore/app/viewsets/repository.py

```diff
@@ -140,6 +140,19 @@ class PublisherViewSet(NamedModelViewSet):
     queryset = Publisher.objects.all()
     filter_class = PublisherFilter
 
+    def update(self, request, *args, **kwargs):
+        partial = kwargs.pop('partial', True)
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=request.data, partial=partial)
+        serializer.is_valid(raise_exception=True)
+        async_result = tasks.publisher.update.apply_async_with_reservation(
+            tags.RESOURCE_REPOSITORY_TYPE, instance.repository.name,
+            kwargs={'publisher_id': instance.id,
```

Comment:
publisher_id and data should be passed with a args keyword argument (since that's what pulpcore.app.tasks.publisher.update() is expecting)

```
tasks.publisher.update.apply_async_with_reservation(
    tags.RESOURCE_REPOSITORY_TYPE, instance.repository.name,
    args=(publisher_id, data, ),
    kwargs={'partial': partial}
)
```